sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -895,6 +895,24 @@ class JDBCSuite extends QueryTest
       "BIT")
     assert(msSqlServerDialect.getJDBCType(BinaryType).map(_.databaseTypeDefinition).get ==
       "VARBINARY(MAX)")
+    assert(msSqlServerDialect.getJDBCType(ByteType).map(_.databaseTypeDefinition).get ==
+      "TINYINT")
+
+    assert(msSqlServerDialect.getJDBCType(ShortType).map(_.databaseTypeDefinition).get ==
+      "SMALLINT")
+  }
+
+  test("MsSqlServerDialect catalyst type mapping") {
```

Comment:
It is best to add tests in [`MsSqlServerIntegrationSuite`](https://github.com/apache/spark/blob/master/external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSqlServerIntegrationSuite.scala) as well.

Reply:
@wangyum Thanks for pointing me to MsSqlServerIntegration.  Agree that's a good place to add these test. Looks like these test spin up a SQL server for test. Is this run in CI? Any pointers to how to run these test.

Reply:
```
./build/mvn install -DskipTests
./build/mvn test -Pdocker-integration-tests -pl :spark-docker-integration-tests_2.12
```

http://spark.apache.org/docs/latest/building-spark.html#running-docker-based-integration-test-suites

Reply:
Thanks

Reply:
@wangyum Thanks for this comment. Integration test brought to light another issue that i have now fixed in JDBCutils.scala (Byte was being written as Int, thus Byte with value 255 was interpreted as -1, resulting in a JDBC exception).